src/apps/ENGINE/src/data.cpp

```diff
@@ -51,6 +56,40 @@ DATA::DATA(DATA &&data) noexcept(false)
     data.AttributesClass = nullptr;
 }
 
+DATA & DATA::operator=(const DATA &data)
+{
+    if (this == &data)
+    {
+        return *this;
+    }
+
+    Data_type = data.Data_type;
+    Number_of_elements = data.Number_of_elements;
+    lValue = data.lValue;
+    fValue = data.fValue;
+    sValue = _strdup(data.sValue);
```

Comment:
_strdup calls malloc, while the memory is freed by delete, not free (btw not correct, should be delete[]). Are you sure it's a good practice to mix malloc and new and free both with one operator?

Reply:
Nice catch! I think we need to have storm_strdup at the moment. Because, there are a few similar places with _strdup/delete.
It seems that msvc malloc is smart enough, but I don't think we are gonna stay with malloc or at least stock malloc for a long time.

Reply:
also, I think _strdup is not cross-platform.

Reply:
Actually, in POSIX we have `strdup` :) Windows implements a lot of POSIX functions prefixing them with an underscore. 
Also, there are too many non-cross-platform things anyway.
But this is of course does not justify using platform-specific code while refactoring...

Honestly, I think more and more about implementing our domain-specific containers compatible with STL yet performance-friendly: nullable string, static string(probably we can steal it from boost), vector with bare resize (not calloc-like), etc.
I would just replace it for std::string until that but overhead might be noticeable since this is a kind of engine primitive that must remain as fast as possible. So I will probably stick with impkementing storm_strdup for now

Reply:
I think it's totally fine to rely on the standard library containers and then replace them in certain places where _profiling_, not our intuition, shows bottlenecks. 
I don't think we're there yet to pull abseil/folly or what's worse implement our own "efficient" standard containers.

Reply:
This is not always true... The problem is that such modifications tends to decrease performance a little bit. Normally you won't notice e.g. 10% performance drop when replacing all c-primitives to c++, but here this will directly impact fps. And this won't be a bottle neck because the whole STL will be a bottle neck, so you won't be able to just profile that as regular code. That's why it's a bit tricky to replace such places with STL and I don't think this is a good idea
P.S. this is only about high-loaded primitives, e.g. scripts compiler entities that may get initialized and destroyed thousands and thousands of times within a second, not about the whole engine code, of course

Reply:
Why wouldn't a profiler be able to show a bottleneck, say, in `DATA::operator=(const DATA &data)`?
I'm just afraid we're talking about premature optimizations which are not even backed up by any profiling results.

Reply:
This is not an easy question 🤔 

IMO, since DATA (and some other primitives) is heavily used in each frame, avoiding overhead in such a plain entity would be a reasonable action, not premature optimization. But this is not the only reason.

Engines specifics are not always about bottlenecks. Imagine you have 100 different entities(not in terms of SE entities) that use 0.1% cpu time each = 10% in total. After modifying them, e.g. by changing something inside, they will use 0.2% each = 20% in total. There still more heavy entities in terms of self-cpu that could look like bottlenecks, but the real reason of 10% CPU drop would lay in these inconspicuous entities. So this is also about choosing an approach, not just local optimization, since we have many similar usages in the code. 

Now imagine another case, you use malloc allocator (it doesn't matter whether it's provided by standard library or not). Potentially, that code may do thousands of unwanted allocations with `std::string`. Many profilers (including one from msvc toolset) does not count kernel time as self-cpu hence you won't even see the difference in the CPU, though you may feel the difference in game.

So it's a bit tricky, but I would go with proper solution right away. Do you have any specific against that?

Reply:
Don't really want to start arguing about unrelated topics in the PR, so feel free to do as you see fit. These were just my two cents, it's not a big deal after all. :)